### PR TITLE
feat: support hostnetwork

### PR DESCRIPTION
**Pull request by fengxsong**
1. define `podTemplate` function;
2. fix `enable_cpu_affinity`;
3. `pluginAttrs` now can merge with other attributes of base plugins like `prometheus`  and customize plugins

### CODE REVIEW

**Comment by tao12345666333** [comment]
sorry for delay. 

could you please resove the conflicts? thansk!

**Comment by fengxsong** [comment]
@tao12345666333 conflicts resolved.

File: charts/apisix/templates/_helpers.tpl [hunk 1]
```diff
@@ -74,3 +74,39 @@ Usage:
...
+  requiredDuringSchedulingIgnoredDuringExecution:
+    - labelSelector:
+        matchExpressions:
+          - key: app.kubernetes.io/instance
+            operator: In
+            values:
+              - {{ .Release.Name }}
+      topologyKey: "kubernetes.io/hostname"
+{{- end }}
+{{- end -}}
```
Comment (tokers) [inline charts/apisix/templates/_helpers.tpl:112]: Add the EOL symbol.

File: charts/apisix/templates/_pod.tpl [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  nodeSelector:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  affinity:
+  {{- merge .Values.apisix.affinity (include "apisix.podAntiAffinity" . | fromYaml) | toYaml | nindent 4 }}
+  {{- with .Values.apisix.tolerations }}
+  tolerations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+{{- end -}}
```
Comment (tokers) [inline charts/apisix/templates/_pod.tpl:176]: Ditto.

File: charts/apisix/templates/configmap.yaml [hunk 3]
```diff
@@ -166,7 +166,7 @@ data:
       error_log: "{{ .Values.logs.errorLog }}"
       error_log_level: "{{ .Values.logs.errorLogLevel }}"         # warn,error
       worker_processes: "{{ .Values.nginx.workerProcesses }}"
-      enable_cpu_affinity: {{ default "true" .Values.nginx.enableCPUAffinity }}
+      enable_cpu_affinity: {{ and true .Values.nginx.enableCPUAffinity }}
```
Comment (tokers) [inline charts/apisix/templates/configmap.yaml:169]: Is the `and` for avoiding a missing `enableCPUAffinity`

**Review by tokers — COMMENTED** [review]
Nice PR!

File: charts/apisix/templates/configmap.yaml (hunk 3 above)
Comment (fengxsong) [reply charts/apisix/templates/configmap.yaml:169]: `default` tempalte function will make `enable_cpu_affinity` property always been `true`.

**Review by tokers — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**